model/project_ref.go

```diff
@@ -1106,19 +1110,38 @@ func GetTasksWithOptions(projectName string, taskName string, opts GetProjectTas
 		task.DisplayNameKey: taskName,
 		task.StatusKey:      bson.M{"$in": finishedStatuses},
 	}
+	var repo *Repository
+	var revisionMatch []bson.M
+	if opts.NumVersions > 0 {
+		repo, err = FindRepository(projectId)
+		if err != nil {
+			return nil, err
+		}
+		if repo == nil {
+			return nil, errors.Errorf("finding repository '%s'", projectId)
+		}
+		revisionMatch = []bson.M{
+			{task.RevisionOrderNumberKey: bson.M{"$gte": repo.RevisionOrderNumber - opts.NumVersions}},
```

Comment:
NumVersions has since been removed and after the most recent refactor the revision numbers will be bound starting either at the repo's revision number or `StartAt` if the user defined it. In both cases the query will not look back more than 1000 revisions